src/main/java/com/bandwidth/sdk/messaging/BandwidthMessagingException.java

```diff
@@ -0,0 +1,49 @@
+package com.bandwidth.sdk;
+
+
+public class BandwidthMessagingException extends RuntimeException {
+
+    private static final String MESSAGE_ERROR_TYPE = "message-failed";
+    
+    private int errorCode;
+    private MessageEvent message;
+
+
+    public BandwidthMessagingException(String message) {
+        super(message);
+    }
+
+    public BandwidthMessagingException(Throwable cause) {
+        super(cause);
+    }
+
+    public BandwidthMessagingException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    public BandwidthMessagingException(String message, int errorCode, Throwable cause) {
+        super(message, cause);
+        this.errorCode = errorCode;
+    }
+
+    public BandwidthMessagingException(String message, MessageEvent messageEvent, Throwable cause) {
+        super(message, cause);
+        this.messageEvent = messageEvent;
+        if (messageEvent.getErrorCode().isPresent()) { //no reason this should not be the case
+            this.errorCode = messageEvent;
+        }
+    }
+
+    /**
+     * Specifics for error codes can be found here https://dev.bandwidth.com/v2-messaging/codes.html
+     */
+    public MessageErrorTypes getErrorType(){
```

Comment:
I don't think `MessageEvent` specific types belong here. You can obtain a `MessageEvent` from parsing a callback, and that already has helper methods for the error type. The first 3 constructors are useful though

Reply:
Gotcha. So we don't need to keep the errorCode at all then